psa-crypto/src/types/key_derivation_operation.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # PSA Operation types
+
+use super::algorithm::{Hash, KeyDerivation};
+use super::key::Id;
+#[cfg(feature = "interface")]
+use super::status::{Error, Result, Status};
+#[cfg(feature = "interface")]
+use core::convert::{From, TryFrom};
+
+/// Key derivation operation for deriving keys from existing sources
+#[derive(Debug, Clone)]
+pub struct KeyDerivationOperation<'a> {
```

Comment:
Still about naming: would it be nicer if this file is named key_derivation and this structure Operation?

Reply:
I think so, the shorter file name still conveys what's there. Just calling it `Operation` _could_ be confusing for people if they have multiple different types of `Operation` in the same place, but they can just rename it themselves in the import statement.